### ISSUE TITLE
Minor version upgrades don't update the `main` Dockerfile

### ISSUE THREAD

**Issue by SuperQ** [prometheus/golang-builder#235]
It seems like there is a bug in the updater tool that misses updating the version of the base FROM in the main Dockerfile.

See: https://github.com/prometheus/golang-builder/pull/234

**Comment by mahendrapaipuri** [comment]
@SuperQ Seems like the image has not been updated on `quay`. I did a test this morning and I still have `go 1.19`:

```
docker run -it --rm --entrypoint /bin/bash quay.io/prometheus/golang-builder:1.21-main
Unable to find image 'quay.io/prometheus/golang-builder:1.21-main' locally
1.21-main: Pulling from prometheus/golang-builder
6a70103cc499: Pull complete 
efce70b838f7: Pull complete 
b540855c13b1: Pull complete 
d7c10a3f17e8: Pull complete 
1d9c9bee2c70: Pull complete 
2dbd3e101c65: Pull complete 
83ef3d7b1838: Pull complete 
abdc1126c656: Pull complete 
d3662ceb12ff: Pull complete 
4f4fb700ef54: Pull complete 
17efe118d3a0: Pull complete 
8ea1d3258d40: Pull complete 
Digest: sha256:028f99b917318664f46b28d3e993b3023bdd30fbe4ecabba511646a65f622b58
Status: Downloaded newer image for quay.io/prometheus/golang-builder:1.21-main
root@415e34401e1a:/app# go version
go version go1.19.12 linux/amd64
```

I see that Circle CI has published the images to the registry following #234 merge. This is strange. Could you take a look at it if you got time?

Cheers!

**Comment by SuperQ** [comment]
Ahh, it looks like I didn't wait enough time for the previous PR merge (https://github.com/prometheus/golang-builder/pull/233) to finish before merging the next one. The second build completed faster than the first.

The nightly auto-publish will happen shortly and should correct the latest version.
